### PR TITLE
Add Initial Cooldown To Communications Computers

### PR DESCRIPTION
**Pull request by Jrpl**
## About the PR
Added an initial announcement cooldown to communications consoles when they're initialized.

## Why / Balance
https://github.com/space-wizards/space-station-14/issues/8506

## Technical details
- Created a new variable on CommunicationsConsoleComponent to track InitialDely
- Set the AnnouncementCooldownRemaining on the console component to the InitialDelay when the console is initialized.

## Media
https://github.com/space-wizards/space-station-14/assets/30051988/26728d44-5311-4bc9-8c60-c0c677d05432

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**
:cl:
- tweak: The Communications Computer now has a 30-second delay before an announcement can be made, after the computer is built. This includes all Communications Computers that are on the map by default.

### CODE REVIEW

**Review by EmoGarbage404 — CHANGES_REQUESTED** [review]
you could skip all of the code and make this pr one line long by just setting the default value of AnnouncementCooldownRemaining to 30 instead of implementing any of the logic.

**Comment by Jrpl** [comment]
@EmoGarbage404 The goal of this change is not to adjust the default 90 second announcement cooldown. When you unscrew and screw a communications computer it resets the announcement delay so you can spam announcements. With this change, I added an InitialDelay whenever a communications computer is initialized. So even if you unscrew and screw a communications computer, you still have to wait at minimum 30 seconds before doing an announcement. After the 30 second delay, it goes back to being the normal 90 second delay between announcements.

**Comment by metalgearsloth** [comment]
Why wouldn't the initial delay match the delay?

File: Content.Server/Communications/CommunicationsConsoleSystem.cs [hunk 1]
```diff
@@ -52,6 +52,9 @@ public override void Initialize()
             SubscribeLocalEvent<CommunicationsConsoleComponent, CommunicationsConsoleAnnounceMessage>(OnAnnounceMessage);
             SubscribeLocalEvent<CommunicationsConsoleComponent, CommunicationsConsoleCallEmergencyShuttleMessage>(OnCallShuttleMessage);
             SubscribeLocalEvent<CommunicationsConsoleComponent, CommunicationsConsoleRecallEmergencyShuttleMessage>(OnRecallShuttleMessage);
+
+            // On console init, set cooldown
+            SubscribeLocalEvent<CommunicationsConsoleComponent, ComponentStartup>((uid, comp, _) => OnCommunicationsConsoleInit(comp));
         }
```
Comment (metalgearsloth) [inline Content.Server/Communications/CommunicationsConsoleSystem.cs:58]: This breaks with serialization as it gets bulldozed every time, use timeoffsetserializer.

**Review by metalgearsloth — CHANGES_REQUESTED** [review]

**Comment by Jrpl** [comment]
> Why wouldn't the initial delay match the delay?

The only reason I made them separate was so you didn't have to wait 90 seconds on round start before doing announcement. We can make them the same if you feel it's best.

File: Content.Server/Communications/CommunicationsConsoleSystem.cs (hunk 1 above)
Comment (Jrpl) [reply Content.Server/Communications/CommunicationsConsoleSystem.cs:58]: Didn't know this existed, thank you! I'm refactoring now. I think I can remove the side effect of not being able to announce immediately on map init.

Comment (Jrpl) [reply Content.Server/Communications/CommunicationsConsoleSystem.cs:58]: After looking into this, I think the Comms Console needs to be refactored for the TimeOffSetSerializer and TimeSpan to work properly with it. I wasn't able to get it working in its current state. I will keep looking into it.

**Review by metalgearsloth — APPROVED** [review]

**Pull request merged**